src/runtime/hexagon/hexagon_power_manager.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "hexagon_power_manager.h"
+
+#include <AEEStdDef.h>
+#include <AEEStdErr.h>
+
+#include "HAP_power.h"
+#include "hexagon_common.h"
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+
+HexagonPowerManager::HexagonPowerManager() {
+  hap_pwr_ctx_ = HAP_utils_create_context();
+  PowerOnHVX();
+  PowerOnHTP();
+  SetAppType();
+  SetDCVS();
+}
+
+HexagonPowerManager::~HexagonPowerManager() {
+  PowerOffHTP();
+  PowerOffHVX();
+  HAP_utils_destroy_context(hap_pwr_ctx_);
+}
+
+void HexagonPowerManager::PowerOnHVX() {
```

Comment:
Can different HVX instances be powered on and off separately?

Reply:
Nope. It's all or nothing.

I'm not actually sure this does anything, but it *might* reduce the latency hit upon executing the first HVX instruction.